Update the Linux pages (#1307)
privacyguides/privacyguides.org: docs/linux-desktop/hardening.en.md

- [Recommended boot parameters](https://madaidans-insecurities.github.io/guides/linux-hardening.html#boot-parameters)
- [Additional recommendations to reduce the kernel's attack surface](https://madaidans-insecurities.github.io/guides/linux-hardening.html#kernel-attack-surface-reduction)

> Do **not** disable unprivileged user namespaces if you use products that rely on it like Podman, Docker, LXC containers, Chromium browser. This option will prevent this software from working.

Reviewer: ```suggestion
Do **not** disable unprivileged user namespaces if you use software that relies on it, like: Podman, Docker, LXC containers, Chromium browser. The option will prevent this software from working.
```

## Linux-Hardened

Some distributions like Arch Linux have the [linux-hardened](https://github.com/anthraxx/linux-hardened), kernel package. It includes [hardening patches](https://wiki.archlinux.org/title/security#Kernel_hardening) and more security-conscious defaults. Linux-Hardened has `kernel.unprivileged_userns_clone=0` disabled by default. See the [warning above](#kernel-hardening) about how this might impact you.

## Linux Kernel Runtime Guard (LKRG)

> LKRG is a kernel module that performs runtime integrity check on the kernel to help detect detect exploits against the kernel. LKRG works in a *post*-detect fashion meaning that it attempts to respond to unauthorized modifications to the running Linux kernel. The intention is that it will defeat many pre-existing exploits as well as undiscovered vulnerabilities. Some hardened distributions [such as Kicksecure](https://www.kicksecure.com/wiki/Linux_Kernel_Runtime_Guard_LKRG) have documentation.

Reviewer: ```suggestion
LKRG is a kernel module that performs runtime integrity check on the kernel to help detect detect exploits against the kernel. LKRG works in a *post*-detect fashion, meaning that it attempts to respond to unauthorized modifications to the running Linux kernel. The intention is that it will defeat many pre-existing exploits as well as undiscovered vulnerabilities. Some hardened distributions, such as [Kicksecure](https://www.kicksecure.com/wiki/Linux_Kernel_Runtime_Guard_LKRG), have documentation.
```

## GRSecurity

> GRSecurity is a set of kernel patches that attempt to improve security of the Linux kernel. While open source  access requires [subscription to a paid model](https://grsecurity.net/passing_the_baton).

Reviewer: ```suggestion
GRSecurity is a set of kernel patches that attempt to improve security of the Linux kernel. Open source  access, however, requires [subscription to a paid model](https://grsecurity.net/passing_the_baton).
```

## Simultaneous multithreading (SMT)

[SMT](https://en.wikipedia.org/wiki/Simultaneous_multithreading) has been the cause of numerous hardware vulnerabilities, and subsequent patches for those vulnerabilities often come with performance penalties that negate most of the performance gain given by SMT. If you followed the “kernel hardening” section above, some kernel parameters already disable SMT. If the option is available to you, we recommend that you disable it in your firmware as well.

## Hardened memory allocator

> The [hardened memory allocator](https://github.com/GrapheneOS/hardened_malloc) from [GrapheneOS](https://grapheneos.org) can be used on general Linux distributions. It is available (though not enabled by default) on Whonix and Kicksecure. It is also available as an [AUR package](https://wiki.archlinux.org/title/Security#Hardened_malloc) on Arch based distributions. If you are using Whonix, Kicksecure or the AUR package, consider setting up `LD_PRELOAD` as described in the [Kicksecure Documentation](https://www.kicksecure.com/wiki/Hardened_Malloc) or [Arch Wiki](https://wiki.archlinux.org/title/Security#Hardened_malloc).

Reviewer: ```suggestion
The [hardened memory allocator](https://github.com/GrapheneOS/hardened_malloc) from [GrapheneOS](https://grapheneos.org) can also be used on general Linux distributions. It is available as an [AUR package](https://wiki.archlinux.org/title/Security#Hardened_malloc) on Arch based distributions, and (though not enabled by default) on Whonix and Kicksecure.

If you are using Whonix, Kicksecure or the AUR package, consider setting up `LD_PRELOAD` as described in the [Kicksecure Documentation](https://www.kicksecure.com/wiki/Hardened_Malloc) or [Arch Wiki](https://wiki.archlinux.org/title/Security#Hardened_malloc).
```

## Linux-Hardened

Some distributions like Arch Linux have the [linux-hardened](https://github.com/anthraxx/linux-hardened), kernel package. It includes [hardening patches](https://wiki.archlinux.org/title/security#Kernel_hardening) and more security-conscious defaults. Linux-Hardened has `kernel.unprivileged_userns_clone=0` disabled by default. See the [warning above](#kernel-hardening) about how this might impact you.

## Linux Kernel Runtime Guard (LKRG)

LKRG is a kernel module that performs runtime integrity check on the kernel and help detection of exploits against the kernel. While [bypassable by design](https://lkrg.org/), LKRG does stop off-the-shelf malware and makes developing exploits harder. If you can get LKRG on your system without too much effort, we recommend that you install it.

Whonix, Kicksecure and Debian based operating systems can get LKRG DKMS by following this [guide](https://www.kicksecure.com/wiki/Linux_Kernel_Runtime_Guard_LKRG).

> ## GRSecurity
> 
> GRSecurity is a set of kernel patches that improves kernel security. Unfortunately, they went [closed](https://grsecurity.net/passing_the_baton) in 2017, and users must purchase access to new versions of their patchset. It is highly recommended that you get GRSecurity if it is available to you. 
> 
> 

Reviewer 2: ```suggestion
```
Really funny they call themselves "Open Source Security, Inc."

I'd shy away from making this recommendation again for the reasons @dngray linked. Their code quality seems [questionable](https://twitter.com/marcan42/status/724745886794833920), and a supposed security company intentionally making things as difficult/annoying as possible to have their patches merged upstream leaves a bad taste in my mouth. It's like the opposite of teams like GrapheneOS, big yikes (lol)

Reviewer 3: No they aren't closed source [guessing that's what you meant?], they just only offer the service to paid clients. The code is still GPL. Their contributions to modern exploit mitigations have made a significant impact on the security of the entire world. It's not their fault that upstream Linux is actively hostile to security researchers and that companies were using their test patches instead of paying for a subscription.

Reviewer 4: Not opposed to mentioning it because it has a subscription model, though IP blocking anyone who criticizes code quality seems a bit unprofessional.

I have to wonder how common grsecurity is, these days. Nowhere I've been uses it.

Reviewer 5: Honestlt i don't think its our place to go into politics like this. We should be focusing on recommending the best options out there and education, not foss politics and drama.

I am personally in favor of adding GRsecurity.
